### DISCUSSION TITLE
INAV 5.1 freezes intermittently on H743-WING V3

### DISCUSSION THREAD

**Discussion by Ktr128** [iNavFlight/inav#8554] [Q&A]
Matek H743-WING V3, INAV 5.1.
The elevon servos hung during autostart and the fixed wing crashed. At the same time, when I saw that the wing was taking off incorrectly, I tried to fix it with sticks, but there was no reaction. Although before that it took off perfectly 3 times, flew missions and returned home.
After that, I caught elevons hanging when testing in my hand.
Turned on the ANGLE mode and moved the flying wing in pitch.
After about 30 seconds, the elevons hung for 1-2 seconds, after which they started working again.

MG90S servos, flysky ia6b receiver are OK. Used this wing for over one year with no problems.

All connections and settings were made in accordance with the instructions:
http://www.mateksys.com/?portfolio=h743-wing-v2

I also use matek f722-wing and sp3 racing f3 deluxe. They never had such problems.

diff all is attached below.
Can you please tell me what is causing these freezes?

```
diff all

# version
# INAV/MATEKH743 5.1.0 Aug 19 2022 / 12:34:02 (76f22b25)
# GCC-10.2.1 20201103 (release)

# start the command batch
batch start

# reset configuration to default settings
defaults noreboot

# resources

# mixer

mmix reset

mmix 0  1.000  0.000  0.000  0.000

# servo mixer
smix reset

smix 0 2 0 -60 0 -1
smix 1 2 1 40 0 -1
smix 2 3 0 60 0 -1
smix 3 3 1 40 0 -1

# servo
servo 1 1461 2000 1500 100
servo 2 1000 1900 1450 100
servo 3 1000 1950 1400 -100

# safehome

# feature
feature MOTOR_STOP
feature GPS
feature PWM_OUTPUT_ENABLE
feature FW_LAUNCH
feature FW_AUTOTRIM

# beeper

# blackbox
blackbox -NAV_ACC
blackbox NAV_POS
blackbox NAV_PID
blackbox MAG
blackbox ACC
blackbox ATTI
blackbox RC_DATA
blackbox RC_COMMAND
blackbox MOTORS

# map

# serial
serial 1 2 115200 115200 0 115200

# led

# color

# mode_color

# aux
aux 0 0 0 1725 2075
aux 1 1 2 1725 2100
aux 2 12 2 900 1075
aux 3 10 1 1750 2100
aux 4 28 6 1750 2100
aux 5 21 3 1725 2100
aux 6 54 4 1725 2100

# adjrange

# rxrange

# temp_sensor

# wp
#wp 0 invalid

# osd_layout

# logic
logic 0 1 -1 1 2 18 0 1 0
logic 1 1 0 38 0 4 0 1700 0

# global vars

# programmable pid controllers

# master
set gyro_main_lpf_hz = 25
set dynamic_gyro_notch_q = 250
set dynamic_gyro_notch_min_hz = 30
set gyro_zero_x = 6
set gyro_zero_y = 13
set gyro_zero_z = -1
set ins_gravity_cmss =  981.697
set acc_hardware = ICM42605
set acczero_x = -5
set acczero_y = -3
set acczero_z = 5
set accgain_x = 4093
set accgain_y = 4090
set accgain_z = 4100
set align_mag = CW270FLIP
set mag_hardware = NONE
set baro_hardware = SPL06
set serialrx_provider = IBUS
set motor_pwm_protocol = STANDARD
set failsafe_procedure = RTH
set align_board_yaw = 2700
set platform_type = AIRPLANE
set model_preview_type = 8
set small_angle = 180
set imu_acc_ignore_rate = 9
set imu_acc_ignore_slope = 5
set applied_defaults = 3
set gps_sbas_mode = AUTO
set gps_ublox_use_galileo = ON
set airmode_type = STICK_CENTER_ONCE
set inav_w_z_baro_p =  0.350
set nav_wp_radius = 5000
set nav_wp_safe_distance = 65000
set nav_rth_allow_landing = NEVER
set nav_rth_altitude = 8000
set nav_rth_home_altitude = 8000
set nav_fw_bank_angle = 45
set nav_fw_loiter_radius = 9000
set nav_fw_control_smoothness = 2
set nav_fw_launch_max_altitude = 8000

# profile
profile 1

set fw_p_pitch = 15
set fw_i_pitch = 5
set fw_d_pitch = 5
set fw_ff_pitch = 92
set fw_p_roll = 15
set fw_i_roll = 3
set fw_d_roll = 7
set fw_ff_roll = 58
set fw_p_yaw = 20
set fw_i_yaw = 0
set fw_ff_yaw = 100
set max_angle_inclination_rll = 550
set dterm_lpf_hz = 10
set fw_turn_assist_pitch_gain =  0.200
set nav_fw_pos_z_p = 15
set nav_fw_pos_z_d = 5
set nav_fw_pos_xy_p = 60
set d_boost_min =  1.000
set d_boost_max =  1.000
set rc_expo = 30
set rc_yaw_expo = 30
set roll_rate = 21
set pitch_rate = 14
set yaw_rate = 3

# profile
profile 2


# profile
profile 3


# battery_profile
battery_profile 1

set throttle_idle =  5.000
set nav_fw_cruise_thr = 1700
set nav_fw_min_thr = 1000
set nav_fw_max_thr = 2000

# battery_profile
battery_profile 2


# battery_profile
battery_profile 3


# restore original profile selection
profile 1
battery_profile 1

# save configuration
save

# 
```

**Comment by EhAye** [comment]
try disabling Servo Autotrim in the configuration tab. There's a known but and they're working on it

If you search h743 in the issues tab you'll see a few people have had this problem

**Reply by Ktr128** [reply]
Thanks for the idea!
You meant #8424 ?

**Comment by hidmax** [comment]
Matek H743-WING V3, INAV 5.1. 
I ran into the same problem! I have a sticky elevator on my plane. Delay for 1-2 seconds, especially noticeable in extreme positions. I don't use Autotrim.

**Comment by Ktr128** [comment]
I disabled auto trim and blackbox.
8 flights so far so good.

**Reply by MrD-RC** [reply]
Interesting. What happens if continuous servo trim is turned back on? Maybe this could be down to blackbox.

**Reply by Jetrell** [reply]
> I disabled auto trim and blackbox. 8 flights so far so good.

Without writing a big spiel.. I have reason to think its related to the blackbox logger `SDIO` driver on the Matek H743 and the F765. 
While other targets run `SPI`, 
The effect seems to manifest, by delaying other tasks, when the logger is enabled, together with slower PWM motor drive protocols. Depending on what other functions you have enabled.

1) It can slow or delay the servo's. 
2) Slow the OSD refresh rate.
3) Reboot the ESC in flight, for no apparent reason. 

etc etc..

**Reply by Ktr128** [reply]
I didn't have a microsd card inserted. Disabled just in case.

**Reply by Jetrell** [reply]
It doesn't matter if the SD card remains inserted.. As long as the blackbox logging is turned off.

**Reply by b14ckyy** [reply]
this is interesting.
this could explain the stuttering and disconnect randomly when I arm the F765 in the HITL simulation with Blackbox enabled.
I will try to disable to confirm that.

**Reply by Jetrell** [reply]
> this is interesting. this could explain the stuttering and disconnect randomly when I arm the F765 in the HITL simulation with Blackbox enabled. I will try to disable to confirm that.

 I always found the log file became corrupted too. 
Once I started running Dshot300 or 600 with these fixed wing controllers. The problem doesn't seem to manifest anymore.
From what I remember, it started occurred back when Dshot was added to the H743 controllers.. I also think its related to #7759

I'd be interested to hear if other people find the issue goes away. When changing from Dshot150 to Dshot300? With the logging still enabled.

**Reply by b14ckyy** [reply]
The delayed SD card initialization has nothing to do with dshot I guess, as it is a one time issue on fresh cards. The boot delay though, could be. need to check the diffs in that issue thread. 
I run DShot 150 on my XUAV clouds on a F765-WSE with no issues and afaik F765 have only the long init time for fresh cards but not the boot issue that H743 have. 

But I can confirm, that after disabling blackbox on the F765-Wing, the HITL stutter/Disconnect problem is gone. I need to check the clouds if it also has these servo freezes/stutters when arming. Blackbox is enabled there as well but I never noticed before. But usually I just arm, raise the throttle and then launch it. If there is a stutter it had no effect on flight. Logs are also not corrupted at 30Hz logging rate.

**Reply by Jetrell** [reply]
> The delayed SD card initialization has nothing to do with dshot I guess, as it is a one time issue on fresh cards. The boot delay though, could be. need to check the diffs in that issue thread. 

I not entirely sure about that.. I found this lagging of many of the FC functions, was far more pronounced on certain SD card types. Even to the point, the plane could not be flown with that much control lag. It took me a while to figure it out originally. 
While that same card had no issue logging data with an F722 or F405.

**Reply by b14ckyy** [reply]
we have 2 separate issues with SD cards. 
One is the slow first time initialization. 
It takes 10-15min until the FREESPACE file is finally created on a brand new SD card. This happens on F765 but not sure on H743. THis is independent from SD Card type. If the waiting time is over then the FC functions normally. Alternatively you can put the SD in an F722 or F405 FC first, let it initialize (takes a few seconds) and then put it in the F765 and all is fine. 

the other issue is a general compatibility issue with some cards on H743 for sure, no idea about F765 as all my different cards (Intenso, Transcent, SandIsk and Samsung) work in both of my F765 FCs. 

People just constantly mixing up these 2 issues. 

the initialization problem cannot have anything to do with DSHOT as it happens even on a fresh flashed F765 FC where the protocol is set to Multishot by default (and STANDARD on older INAV versions)

**Reply by Jetrell** [reply]
> the initialization problem cannot have anything to do with DSHOT as it happens even on a fresh flashed F765 FC where the protocol is set to Multishot by default (and STANDARD on older INAV versions)

I'm aware of both these issues, and the workarounds for them.

I didn't say that Dshot was the direct cause... But rather, both these issues occurred at the same time that Dshot implementation was added for the H743.
And considering the root cause of either issue has never been solved..  It's not a big stretch to say that its related to the SDIO and DMA timing that was added for H743 use.

Another interesting workaround for this issue, with certain SD cards.
Is to leave the Blackbox enabled, and just turn off the `Motors_output` logging.. My tests showed it also over came the problem with both H743 and F765 wing boards.
